src/main/java/org/glyptodon/guacamole/auth/json/connection/ConnectionService.java

```diff
@@ -205,9 +206,11 @@ public GuacamoleTunnel connect(UserData.Connection connection,
         String hostname = proxyConfig.getHostname();
         int port = proxyConfig.getPort();
 
-        // Generate and verify connection configuration
-        GuacamoleConfiguration config = getConfiguration(connection);
-        if (config == null) {
+        // Apply tokens to config parameters
+        GuacamoleConfiguration filteredConfig = getConfiguration(connection);
+        new TokenFilter(tokens).filterValues(filteredConfig.getParameters());
+
+        if (filteredConfig == null) {
```

Comment:
I think the ordering here is incorrect. If `filteredConfig` is `null`, the attempt to call `filteredConfig.getParameters()` above would have resulted in a `NullPointerException`.